include/lsst/afw/geom/SpanSet.h

```diff
@@ -0,0 +1,252 @@
+    }
+
+    template <typename Functor, typename...T>
+    void applyFunctor(Functor && func, T&&...x) const {
+        /* Use a variadic template to take a functor object, and an arbitrary number
+           of parameters. For each of the arguments, construct a Getter class using
+           a function (makeGetter) which is overloaded to all the types applyFunctorImpl
+           supports: Images, MaskedImages, Exposures, ndarrays, numeric values, and
+           iterators. A note a ndarray can either be interpreted as a two dimensional
+           image, or a one dimensional vector, as such the user must break the ambiguity
+           by using either the ndarray::ndImage or ndarray::ndFlat functions on the array.
+           The functor and the getters are then passed to the implementation of applyFunctor
+           where the values of the input arguments are intelegently generated at each point
+           in SpanSet, and passed to the functor object for evaluation. The functor object
+           must operate inplace. The first two arguments of the functor must be the point
+           in the SpanSet where the operation is occuring and an integer n representing
+           the number of points that have been operated on. i.e. if the SpanSet has an
+           area of 10 pixels, the functor will be passed a value of 9 on the last
+           invocation. All other arguments to the functor are generated from the getters
+           and will be supplied in the order in which they are passed to applyFunctor
+         */
+        applyFunctorImpl(func, details::makeGetter(x)...);
+    }
+
+    template <typename T>
+    void setMask(lsst::afw::image::Mask<T> & target, T bitmask) const {
+        // Use a lambda to set bits in a mask at the locations given by SpanSet
+        auto targetArray = target.getArray();
+        applyFunctor([](Point2I point, int n, typename details::ImageNdGetter<T, 2, 1>::Reference maskVal,
+                        T bitmask){maskVal |= bitmask;}, ndarray::ndImage(targetArray), bitmask);
+    }
+
+    template <typename T>
+    void clearMask(lsst::afw::image::Mask<T> & target, T bitmask) const {
+        // Use a lambda to clear bits in a mask at the locations given by SpanSet
+        auto targetArray = target.getArray();
+        applyFunctor([](Point2I point, int n, typename details::ImageNdGetter<T, 2, 1>::Reference maskVal,
+                        T bitmask){maskVal &= ~bitmask;}, ndarray::ndImage(targetArray), bitmask);
+    }
+
+    // SpanSet functions
+    SpanSet intersect(SpanSet const & other) const;
+    SpanSet intersectNot(SpanSet const & other) const;
+    SpanSet union_(SpanSet const & other) const;
+
+    /// Compareison Operators
+    bool operator==(SpanSet const & other) const;
+    bool operator!=(SpanSet const & other) const;
+
+    static SpanSet spanSetFromShape(int r, Stencil s = Stencil::CIRCLE);
+
+ private:
+    void runNormalize();
```

Comment:
According to the style guideline 3-4, private member functions must be preceded by an underscore.